src/test/java/org/opensearch/security/configuration/ConfigurationRepositoryTest.java

```diff
@@ -133,10 +198,77 @@ public void initOnNodeStart_withSecurityIndexCreationEnabledShouldSetInstallDefa
         ConfigurationRepository configRepository = createConfigurationRepository(settings);
 
         final var result = configRepository.initOnNodeStart();
-
         assertThat(result.join(), is(true));
     }
 
+    @Test
+    public void initOnNodeStart_whenClusterAvailable() {
+        Settings settings = Settings.builder().put(SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true).build();
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+        when(clusterBlocks.hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)).thenReturn(false);
+
+        configRepository.initOnNodeStart().join();
+
+        verify(clusterBlocks, times(1)).hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE);
+    }
+
+    @Test
+    public void initOnNodeStart_whenClusterInitiallyUnavailable() {
+        Settings settings = Settings.builder().put(SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true).build();
+        ConfigurationRepository configRepository = createConfigurationRepository(settings);
+        when(clusterBlocks.hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)).thenReturn(true, true, false);
+
+        configRepository.initOnNodeStart().join();
+
+        verify(clusterBlocks, times(3)).hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE);
+    }
+
+    @Test
+    public void initOnNodeStart_installDefaultConfigSuccess() throws InterruptedException, TimeoutException {
+        // Required for ComplianceConfig in audit.yml to work
+        setupObjectMapperInjectables();
+
+        Settings settings = Settings.builder()
+            .put(SECURITY_ALLOW_DEFAULT_INIT_SECURITYINDEX, true)
+            .put("path.home", TEST_CONFIG_DIR)
+            .put(ConfigConstants.SECURITY_UNSUPPORTED_DELAY_INITIALIZATION_SECONDS, 1)
+            .build();
+        System.setProperty("security.default_init.dir", TEST_CONFIG_DIR);
```

Comment:
Is this necessary? Such global changes can have unexpected side effects on other tests, as these also see that change. This can lead again to test flakiness in case the order of test execution is nondeterministic.

Reply:
yes, this is required to fetch config files used to install the default config. I see your point thought, I can clear the property at end of the test.